src/main/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngines990KnnVectorsReader.java

```diff
@@ -159,4 +194,31 @@ public void close() throws IOException {
     public long ramBytesUsed() {
         return flatVectorsReader.ramBytesUsed();
     }
+
+    private void primeQuantizationStateCache() throws IOException {
+        quantizationStateCacheKeyPerField = new HashMap<>();
+        Map<String, byte[]> stateMap = KNN990QuantizationStateReader.read(segmentReadState);
+        for (Map.Entry<String, byte[]> entry : stateMap.entrySet()) {
```

Comment:
this is causing the BWC to fail. So right now without checking the file is present or not you are reading the QS. In the read function we should first check if file is present or not. Then only we should open the file.

Reply:
```
Caused by: java.nio.file.NoSuchFileException: /home/runner/work/k-NN/k-NN/qa/restart-upgrade/build/testclusters/knnBwcCluster-restart-0/data/nodes/0/indices/7GO1YWCXQt62LsLnXY0JoQ/0/index/_2j_NativeEngines990KnnVectorsFormat_0.osknnqstate
»  	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92) ~[?:?]
»  	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106) ~[?:?]
»  	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111) ~[?:?]
»  	at java.base/sun.nio.fs.UnixFileSystemProvider.newFileChannel(UnixFileSystemProvider.java:224) ~[?:?]
»  	at java.base/java.nio.channels.FileChannel.open(FileChannel.java:309) ~[?:?]
»  	at java.base/java.nio.channels.FileChannel.open(FileChannel.java:369) ~[?:?]
»  	at org.apache.lucene.store.MemorySegmentIndexInputProvider.openInput(MemorySegmentIndexInputProvider.java:58) ~[lucene-core-9.12.0-snapshot-847316d.jar:9.12.0-snapshot-847316d 847316dd1eec2258ccf5da52dca51c01b9df1c4c - 2024-06-27 17:07:28]
»  	at org.apache.lucene.store.MMapDirectory.openInput(MMapDirectory.java:330) ~[lucene-core-9.12.0-snapshot-847316d.jar:9.12.0-snapshot-847316d 847316dd1eec2258ccf5da52dca51c01b9df1c4c - 2024-06-27 17:07:28]
»  	at org.opensearch.index.store.FsDirectoryFactory$HybridDirectory.openInput(FsDirectoryFactory.java:164) ~[opensearch-3.0.0-SNAPSHOT.jar:3.0.0-SNAPSHOT]
»  	at org.apache.lucene.store.FilterDirectory.openInput(FilterDirectory.java:101) ~[lucene-core-9.12.0-snapshot-847316d.jar:9.12.0-snapshot-847316d 847316dd1eec2258ccf5da52dca51c01b9df1c4c - 2024-06-27 17:07:28]
»  	at org.apache.lucene.store.FilterDirectory.openInput(FilterDirectory.java:101) ~[lucene-core-9.12.0-snapshot-847316d.jar:9.12.0-snapshot-847316d 847316dd1eec2258ccf5da52dca51c01b9df1c4c - 2024-06-27 17:07:28]
»  	at org.opensearch.knn.index.codec.KNN990Codec.KNN990QuantizationStateReader.read(KNN990QuantizationStateReader.java:54) ~[?:?]
»  	at org.opensearch.knn.index.codec.KNN990Codec.NativeEngines990KnnVectorsReader.primeQuantizationStateCache(NativeEngines990KnnVectorsReader.java:200) ~[?:?]
»  	at org.opensearch.knn.index.codec.KNN990Codec.NativeEngines990KnnVectorsReader.<init>(NativeEngines990KnnVectorsReader.java:53) ~[?:?]
»  	at org.opensearch.knn.index.codec.KNN990Codec.NativeEngines990KnnVectorsFormat.fieldsReader(NativeEngines990KnnVectorsFormat.java:61) ~[?:?]
»  	at org.apache.lucene.codecs.perfield.PerFieldKnnVectorsFormat$FieldsReader.<init>(PerFieldKnnVectorsFormat.java:222) ~[lucene-core-9.12.0-snapshot-847316d.jar:9.12.0-
```

Reply:
Added

Reply:
Got caught in Unit test only ...It's fixed now.